File: backend/src/main/java/com/affairs/management/entity/Notification.java
package com.affairs.management.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 通知实体
 */
@Data
public class Notification {
    /** 通知ID，如N001 */
    private String id;
    /** 接收用户ID */
    private String userId;
    /** 通知标题 */
    private String title;
    /** 通知内容 */
    private String content;
    /** 通知类型 */
    private String type;
    /** 关联任务ID */
    private String relatedTaskId;
    /** 是否已读：0否 1是 */
    private Integer isRead;
    /** 创建时间 */
    private LocalDateTime createdAt;
}

File: frontend/src/types/notification.ts
// ============================================
// 通知 相关类型
// ============================================

/** 通知类型 */
export type NotificationType =
  | 'task_assigned'    // 事务指派
  | 'task_accepted'    // 事务接收
  | 'task_responded'   // 事务响应
  | 'task_completed'   // 事务完成
  | 'task_submitted'   // 结果提交
  | 'task_approved'    // 审核通过
  | 'task_rejected'    // 审核驳回
  | 'system'           // 系统通知

/** 通知 */
export interface Notification {
  id: string
  userId: string
  title: string
  content: string
  type: NotificationType
  relatedTaskId: string | null
  isRead: boolean
  createdAt: string
}
